scripts/resource/basic_discovery.rc

```diff
@@ -19,6 +19,9 @@ if (framework.datastore['RHOSTS'] == nil)
 	return
 end
 
+#Remove any commas present in the RHOSTS
+framework.datastore['RHOSTS'] = framework.datastore['RHOSTS'].gsub(',','')
```

Comment:
Just to confirm, won't this break things? If there's no spaces present? i.e. `ip1,ip2` will become `ip1ip2` which nmap won't be able to parse

Or does the datastore normalize this value on being set?

Reply:
let me run and check the output once.

Reply:
Also another point we just discussed that I hadn't thought about is that this is modifying datastore options. This may end up introducing bugs if validation takes place in one part of the code and then you update the value here which causes a violation of assumptions down the line. We've seen bugs like this happen in the past. Is there a way we could perhaps modify this closer to where the issue occurs vs modifying the datastore itself?

Reply:
Nmap wont be able to parse it, it should be gsub(',',' ') instead to make IP1,IP2 to IP1 IP2.

In the regular case of IP1, IP2 that would be IP1  IP2, which wouldnt give any problem to nmap.

Reply:
Maybe instead of the datastore mutation, we could add the code change closer to where it needs to be used like these lines:

```
		run_single("db_nmap -v -n #{nmapopts} #{framework.datastore['RHOSTS']}")
```

Reply:
> Maybe instead of the datastore mutation (i.e. the reassignment of RHOSTS), we could add the code change closer to where it needs to be used like these lines:
> 
> ```ruby
> 		run_single("db_nmap -v -n #{nmapopts} #{framework.datastore['RHOSTS'].gsub...etc...}")
> ```

that does make more sense than changing the datastore, let me try this out.

Reply:
> Maybe instead of the datastore mutation (i.e. the reassignment of RHOSTS), we could add the code change closer to where it needs to be used like these lines:
> 
> ```ruby
> 		run_single("db_nmap -v -n #{nmapopts} #{framework.datastore['RHOSTS'].gsub...etc...}")
> ```

This would run the nmap scan without changing the datastore, and since RHOSTS are set for only stage 1, global values remains unchanged. 

I will edit the file to make the necessary changes in this pull request.